### PR TITLE
Make Provider multichain

### PR DESCRIPTION
**Pull request by calina-c**
Works on #572 

Pending TODOs:
- [x] test functions for env var decoding, fallbacks etc.
- [x] return c2d environments indexed by chainId instead of flat list
- [x] update c2d and remove chainId from status, stop and delete payloads
- [x] run a full pre-commit script on all files to fix issues that were allowed during the development phase

### CODE REVIEW

File: ocean_provider/utils/basics.py [hunk 1]
```diff
@@ -32,18 +34,55 @@ def get_config(config_file: Optional[str] = None) -> Config:
...
+        decoded = json.loads(os.environ.get("NETWORK_URL"))
+        return [int(key) for key in decoded.keys()]
+    except (JSONDecodeError, TypeError):
+        web3 = get_web3(os.environ.get("NETWORK_URL"))
+        return [web3.chain_id]
+
+
+def get_provider_addresses():
+    chain_ids = get_configured_chains()
+    if not chain_ids:
```
Comment (calina-c) [inline ocean_provider/utils/basics.py:66]: TODO: there will be one, if NETWORK_URL is singular, it won't be empty

File: ocean_provider/utils/compute_environments.py [hunk 2]
```diff
@@ -19,18 +18,22 @@ def get_c2d_environments() -> List:
...
+        response = requests_session.get(
+            get_compute_environments_endpoint(), headers=standard_headers, params=params
+        )
+
+        # add provider token from config
+        envs = response.json()
+        for env in envs:
+            env["feeToken"] = get_provider_fee_token(chain)
+
+        all_environments.extend(envs)
```
Comment (calina-c) [inline ocean_provider/utils/compute_environments.py:34]: TODO: return indexed by chain instead

**Comment by alexcos20** [comment]
let's test this:

- [ ] create docker image
- [ ] @md00ux - create CNAMEs https://v4-2.provider.NETWORK.oceanprotocol.com that points to https://v4.provider.NETWORK.oceanprotocol.com/  (for all NETWORKs:  mainnet, mumbai, goerli, polygon, bsc, ewf, etc)
- [ ] @md00ux - deploy multichain provider to https://v4.provider.oceanprotocol.com, using private keys from existing providers
- [ ] create market PR that uses new CNAMES for provider URI
- [ ] publish assets (datasets & algorithms), using all known storage types:  url, graphql, ipfs, smartcontract, etc
- [ ] make sure consumption works for all previous created assets
- [ ] once previous steps are done, @md00ux changes all v4-2 CNAMES to multichain provider (IE: https://v4-2.provider.mainnet.oceanprotocol.com/  points to https://v4.provider.oceanprotocol.com
- [ ] make sure consumption works for all previous created assets
- [ ] update market PR to use https://v4.provider.oceanprotocol.com as provider URI for all networks
- [ ] publish another set of assets (datasets & algorithms), using all known storage types:  url, graphql, ipfs, smartcontract, etc
- [ ] make sure consumption works for all previous created assets

**Comment by AnaLoznianu** [comment]
Moved the testing part of Multichain Provider to this issue: https://github.com/oceanprotocol/provider/issues/591

**Comment by mihaisc** [comment]
@calina-c can you please do a rebase so we can test this

**Review by mihaisc — APPROVED** [review]

**Pull request merged**